apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -138,7 +133,60 @@ export const Lti13ClaimSchema = z.object({
   // https://www.imsglobal.org/spec/lti/v1p3#vendor-specific-extension-claims
   // My development Canvas sends their own named extension as a top level property
   // "https://www.instructure.com/placement": "course_navigation"
+
+  // https://www.imsglobal.org/spec/lti-ags/v2p0#assignment-and-grade-service-claim
+  'https://purl.imsglobal.org/spec/lti-ags/claim/endpoint': z
+    .object({
+      lineitems: z.string().optional(),
+      lineitem: z.string().optional(),
+      scope: z.string().array(),
+    })
+    .optional(),
+
+  // https://www.imsglobal.org/spec/lti-nrps/v2p0/#resource-link-membership-service
+  'https://purl.imsglobal.org/spec/lti-nrps/claim/namesroleservice': z
+    .object({
+      context_memberships_url: z.string(),
+      service_versions: z.literal('2.0').array(),
+    })
+    .optional(),
 });
+
+// https://www.imsglobal.org/spec/lti/v1p3#required-message-claims
+export const Lti13ResourceLinkRequestSchema = Lti13ClaimBaseSchema.merge(
+  z.object({
+    'https://purl.imsglobal.org/spec/lti/claim/message_type': z.literal('LtiResourceLinkRequest'),
+    'https://purl.imsglobal.org/spec/lti/claim/resource_link': z.object({
+      id: z.string(),
+      description: z.string().nullish(),
+      title: z.string().nullish(),
+    }),
+  }),
+).passthrough();
+
+// https://www.imsglobal.org/spec/lti-dl/v2p0#message-claims
+export const Lti13DeepLinkingRequestSchema = Lti13ClaimBaseSchema.merge(
+  z.object({
+    'https://purl.imsglobal.org/spec/lti/claim/message_type': z.literal('LtiDeepLinkingRequest'),
+    'https://purl.imsglobal.org/spec/lti-dl/claim/deep_linking_settings': z.object({
+      deep_link_return_url: z.string(),
+      accept_types: z.string().array(),
```

Comment:
I love that the spec doesn't tell us what values will be in this array 😂

Reply:
I think they infer it from https://www.imsglobal.org/spec/lti-dl/v2p0#content-item-types  I'll switch our schema to an enum.

Reply:
> The type is open for extension; it is expected other IMS Global specifications as well as platforms will extend on the core set of types defined in this document.

This makes me wary of using an enum. I don't want us to error out if we get some unknown value, we should just ignore it, right?

Reply:
That's fair. Switched back to `z.string().array()` in 0b106cb.